dev/storage.h

```diff
@@ -1439,43 +1439,8 @@ namespace sqlite_orm {
                     }
                     ss << " ";
                 }
-                auto query = ss.str();
-                sqlite3_stmt *stmt;
-                if (sqlite3_prepare_v2(connection->get_db(), query.c_str(), -1, &stmt, nullptr) == SQLITE_OK) {
-                    statement_finalizer finalizer{stmt};
-                    auto index = 1;
-                    impl.table.for_each_column([&o, stmt, &index] (auto &c) {
-                        if(!c.template has<constraints::primary_key_t<>>()) {
-                            using field_type = typename std::decay<decltype(c)>::type::field_type;
-                            const field_type *value = nullptr;
-                            if(c.member_pointer){
-                                value = &(o.*c.member_pointer);
-                            }else{
-                                value = &((o).*(c.getter))();
-                            }
-                            statement_binder<field_type>().bind(stmt, index++, *value);
-                        }
-                    });
-                    impl.table.for_each_column([&o, stmt, &index] (auto &c) {
-                        if(c.template has<constraints::primary_key_t<>>()) {
-                            using field_type = typename std::decay<decltype(c)>::type::field_type;
-                            const field_type *value = nullptr;
-                            if(c.member_pointer){
-                                value = &(o.*c.member_pointer);
-                            }else{
-                                value = &((o).*(c.getter))();
-                            }
-                            statement_binder<field_type>().bind(stmt, index++, *value);
-                        }
-                    });
-                    if (sqlite3_step(stmt) == SQLITE_DONE) {
-                        //  done..
-                    }else{
-                        throw std::system_error(std::error_code(sqlite3_errcode(connection->get_db()), get_sqlite_error_category()));
-                    }
-                }else {
-                    throw std::system_error(std::error_code(sqlite3_errcode(connection->get_db()), get_sqlite_error_category()));
-                }
+
+                fill_statement(o, ss.str());
```

Comment:
Unfortunately I cannot merge refactoring cause I am going to refactor these functions in another way. I have already implemented it once in a different project which uses custom `sqlite_orm`'s fork. I want to separate ORM and SQLite access logic to erase copy-paste totally and move some implementations to `.cpp` file

Reply:
I suggest use -inl.h pattern for this and I strongly recommend to separate implementation from interface.

but I suggest if this refactor does not bad for you and some time remain to done your project, you can accept this refactor to decreasing code duplication in project, but if you not , I can close pull request

what am I do?

Reply:
I don't think that -inl.h pattern is good. It can be more comfortable for C++ coders but as I said before it is a very popular practice to have implementation in place in different popular languages. I understand that separated implementation reduces compile time for non template classes and I don't say we need to deny it if we would have it but templates have no advantages except readability for some developers. And as I said I cannot merge this refactoring cause I am planing to refactor these functions in a different way.

Reply:
@fnc12 Ok so close this PR if you have planning soon to refactor

Reply:
@fnc12 I close it, but I think it's better to accept it until you refactor whole of `storage` still